src/dev/lupio/LupioRNG.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2021 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from m5.objects.Device import BasicPioDevice
+from m5.params import Param
+from m5.proxy import Parent
+from m5.defines import buildEnv
+from m5.util.fdthelper import FdtPropertyWords
+
+class LupioRNG(BasicPioDevice):
+
+    type = 'LupioRNG'
+    cxx_class='gem5::LupioRNG'
+    cxx_header = 'dev/lupio/lupio_rng.hh'
+    pio_size = Param.Addr(0x1000, "PIO Size")
+    seed = Param.Int(0, "Initial seed for the random number generator")
+    if buildEnv['TARGET_ISA'] == 'riscv':
+        plic = Param.Plic(Parent.any, "PLIC")
+        lupio_rng_int_id = Param.Int(3, "Interrupt ID for the RNG")
```

Comment:
It seems like the interrupt ID should almost always be given by the platform code when building the hardware system. This value is just a default value because we need to give one. In that case, I would suggest that this default value be 0.